plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -416,11 +416,17 @@ public long executeUpdate(ConnectorSession session, QueryJobConfiguration job)
     }
 
     public TableResult executeQuery(ConnectorSession session, String sql)
+    {
+        return executeQuery(session, sql, null);
+    }
+
+    public TableResult executeQuery(ConnectorSession session, String sql, Long maxResults)
     {
         log.debug("Execute query: %s", sql);
         QueryJobConfiguration job = QueryJobConfiguration.newBuilder(sql)
                 .setUseQueryCache(isQueryResultsCacheEnabled(session))
                 .setCreateDisposition(createDisposition(session))
+                .setMaxResults(maxResults)
```

Comment:
The docs say that fast query path depends on the options set on the connection, but doesn't say which ones specifically. I found that setting the job timeout should disable the fast path. I assume if any of the options we use did that, tests would fail. For example, in the query runner, we set `bigquery.rpc-timeout=30s` (the default in config is 0s).